addons/sale_purchase/models/sale_order.py

```diff
@@ -309,8 +309,11 @@ def _purchase_service_generation(self):
         """
         sale_line_purchase_map = {}
         for line in self:
+            # Do not raise an error if there is no seller for the product in another company as service_to_purchase is product linked and not company linked,
+            # using it in interco operations lead to error when the company executing the service is reach through auto-validated SO/PO
+            suppliers = line.company_id == self.env.company or line.product_id._select_seller(quantity=line.product_uom_qty, uom_id=line.product_uom) if line.product_id else False
```

Comment:
I'm not sure to understand the fix. why not avoiding raising the error in `_purchase_service_create` in case of service_to_purchase ?